File: Assignment_29/Program4.c
/*
Write a program which accepts file name from user and one charcter from user and count number of occurence of that Characters from that file
Input : Demo.txt  'A'
Output : Frequency of A is : 5
*/

#include<stdio.h>
#include<stdlib.h>
#include<fcntl.h>
#include<unistd.h>
#include<string.h>

#define BUFFER_SIZE 1024

int CountChar(char FName[],char ch)
{
    int fd = 0;
    char Buffer[BUFFER_SIZE] = {'\0'};   
    int iRet = 0;                             
    int iCnt = 0;
    int iCharCount = 0;

    fd = open(FName,O_RDONLY);         

    if(fd == -1)
    {
        printf("Unable to open File\n");
    }
    else
    {   
        while((iRet = read(fd,Buffer,BUFFER_SIZE)) != 0)
        {
            for(iCnt = 0;iCnt < iRet;iCnt++)
            {
                if(Buffer[iCnt] == ch)
                {
                    iCharCount++;
                }
            }
        }
        close(fd);
    }
    return iCharCount;
}

int main()
{
    char FileName[30];
    int iRet = 0;
    char cValue;

    printf("Enter file name:");
    scanf("%s",FileName);

    printf("Enter the character:");
    scanf(" %c",&cValue);

    iRet = CountChar(FileName,cValue);

    printf("Frequency of Character is : %d",iRet);

    return 0;
}

File: Assignment_56/Program4.java
/* Write a java program which accepts string and one character from user and remove that character from the string.
Input :   IndiaisDemoIndia
          i
Output :  IndasDemoInda
*/

import java.util.*;

class MyString
{
    public void RemoveCharacter(String str,char ch)
    {
        int i = 0;

        String newstr = "";

        for(i = 0;i < str.length();i++)
        {
            if(str.charAt(i) != ch)
            {
                newstr = newstr + str.charAt(i);
            }
        }
        System.out.println(newstr);
    }
}

class Program4
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter the string : ");
        String str = sobj.nextLine();

        System.out.println("Enter the character to remove : ");
        char ch = sobj.next().charAt(0);

        MyString obj = new MyString();
        obj.RemoveCharacter(str,ch);

        sobj.close();
    }
}
